common/rfb/VNCSConnectionST.cxx

```diff
@@ -445,6 +450,11 @@ void VNCSConnectionST::queryConnection(const char* userName)
 
 void VNCSConnectionST::clientInit(bool shared)
 {
+  if (!server->isDesktopReady())
+    return;
```

Comment:
I don't think this is right. If the desktop isn't ready, then the work here is never done.

Nothing here requires the desktop to be ready, so I think this check should just be removed.